src/coreclr/nativeaot/Runtime/threadstore.cpp

```diff
@@ -146,6 +149,21 @@ void ThreadStore::AttachCurrentThread(bool fAcquireThreadStoreLock)
 void ThreadStore::AttachCurrentThread()
 {
     AttachCurrentThread(true);
+
+    // Disallow gcstress on finalizer thread to work around the current implementation's limitation that it will
+    // get into an infinite loop if performed on the finalizer thread.
+    Thread* pAttachedThread = GetCurrentThread();
+    if (pAttachedThread == g_pFinalizerThread)
+    {
+        pAttachedThread->SetSuppressGcStress();
+    }
+
+    // initialize inlined threadstatics (single-module config only)
+    TypeManager* pSingleTypeManager = GetRuntimeInstance()->GetSingleTypeManager();
+    if (pSingleTypeManager != NULL)
+    {
+        InitInlineThreadStatics(pSingleTypeManager);
```

Comment:
This will make thread attach a potential source of unhandled OOM that leads to fail fast. This fail fast is impossible for user code to catch or recover from. I am not sure whether it is a good trade-off to make for native AOT. It will make native AOT less suitable for system-programming like tasks where uncatchable OOMs are a problem.

Reply:
One possible solution is to ignore the OOM on thread attach and live the storage as NULL and turn the AV on the first use of the threadstatic into an OOM exception. That could be inconvenient though when the access is JIT-inlined.

Or we can do one null-check for the whole thing and call the initializer on the first use.